exporters/trace/zipkin/model.go

```diff
@@ -140,27 +147,123 @@ func attributesToJSONMapString(attributes []attribute.KeyValue) string {
 // extraZipkinTags are those that may be added to every outgoing span
 var extraZipkinTags = []string{
 	"otel.status_code",
-	"otel.status_description",
 	keyInstrumentationLibraryName,
 	keyInstrumentationLibraryVersion,
 }
 
 func toZipkinTags(data *export.SpanSnapshot) map[string]string {
 	m := make(map[string]string, len(data.Attributes)+len(extraZipkinTags))
 	for _, kv := range data.Attributes {
-		m[(string)(kv.Key)] = kv.Value.Emit()
+		switch kv.Value.Type() {
+		// For array attributes, serialize as JSON list string.
+		case attribute.ARRAY:
+			json, _ := json.Marshal(kv.Value.AsArray())
+			m[(string)(kv.Key)] = (string)(json)
+		default:
+			m[(string)(kv.Key)] = kv.Value.Emit()
+		}
+	}
+
+	if data.StatusCode != codes.Unset {
+		m["otel.status_code"] = data.StatusCode.String()
+	}
+
+	if data.StatusCode == codes.Error {
+		m["error"] = data.StatusMessage
 	}
+
+	// If boolean with 'false' is present, should be removed.
 	if v, ok := m["error"]; ok && v == "false" {
 		delete(m, "error")
 	}
-	m["otel.status_code"] = data.StatusCode.String()
-	m["otel.status_description"] = data.StatusMessage
 
 	if il := data.InstrumentationLibrary; il.Name != "" {
 		m[keyInstrumentationLibraryName] = il.Name
 		if il.Version != "" {
 			m[keyInstrumentationLibraryVersion] = il.Version
 		}
 	}
+
+	if len(m) == 0 {
+		return nil
+	}
+
 	return m
 }
+
+// Rank determines selection order for remote endpoint. See the specification
+// https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk_exporters/zipkin.md#remote-endpoint
+var remoteEndpointKeyRank = map[attribute.Key]int{
+	semconv.PeerServiceKey: 0,
+	semconv.NetPeerNameKey: 1,
+	semconv.NetPeerIPKey:   2,
+	keyPeerHostname:        3,
+	keyPeerAddress:         4,
+	semconv.HTTPHostKey:    5,
+	semconv.DBNameKey:      6,
+}
+
+func toZipkinRemoteEndpoint(data *export.SpanSnapshot) *zkmodel.Endpoint {
+	// Should be set only for consumer or producer kind
+	if data.SpanKind != trace.SpanKindConsumer &&
```

Comment:
The [specification currently](https://github.com/open-telemetry/opentelemetry-specification/blob/8da2ec13f2b49fb30abd8b05a6eb6166a6e2b597/specification/trace/sdk_exporters/zipkin.md#otlp---zipkin) refers to `SpanKind.CLIENT` rather than `SpanKind.CONSUMER`, but it seems like a typo in the spec to me as well.

Reply:
FYI I opened a [Discussion](https://github.com/open-telemetry/opentelemetry-specification/discussions/1546) in the specification repo to clarify this.

Reply:
Didn't even notice this is the other way around in the spec! I'll keep an eye on that issue, thanks :+1:

Reply:
Left a comment. Overall I believe CLIENT, CONSUMER and PRODUCER support remote endpoint.

Reply:
Understood :+1: I addressed the feedback and stick to the current spec version. After approvals this should be good to go.

Reply:
Tracking in this repo with https://github.com/open-telemetry/opentelemetry-go/issues/1784